internal/core/docker_host.go

```diff
@@ -133,17 +138,14 @@ func extractDockerHost(ctx context.Context) string {
 			continue
 		}
 
-		err = defaultCallbackCheckFn(ctx, dockerHost)
-		if err != nil {
-			outerErr = fmt.Errorf("%w: %w", outerErr, err)
+		if err = defaultCallbackCheckFn(ctx, dockerHost); err != nil {
```

Comment:
suggestion: As this is now returning an error we can make it more useful by filtering out all the not set errors.

Would be nice if all the errors were related so we could use and errors.As which would work without having to add each case which is fragile in the case that more options are set but, this should work for now.
```go
// isHostNotSet returns true if the error is related to the Docker host
// not being set, false otherwise.
func isHostNotFound(err error) bool {
	switch {
	case errors.Is(err, ErrTestcontainersHostNotSetInProperties),
		errors.Is(err, ErrDockerHostNotSet),
		errors.Is(err, ErrDockerSocketNotSetInContext),
		errors.Is(err, ErrDockerSocketNotSetInProperties),
		errors.Is(err, ErrSocketNotFoundInPath),
		errors.Is(err, ErrDockerSocketNotSetInProperties),
		errors.Is(err, ErrXDGRuntimeDirNotSet),
		errors.Is(err, ErrRootlessDockerNotFoundHomeRunDir),
		errors.Is(err, ErrRootlessDockerNotFoundHomeDesktopDir),
		errors.Is(err, ErrRootlessDockerNotFoundRunDir):
		return true
	default:
		return false
	}
}

// extractDockerHost Extracts the docker host from the different alternatives, without caching the result.
// This internal method is handy for testing purposes.
func extractDockerHost(ctx context.Context) (string, error) {
	dockerHostFns := []func(context.Context) (string, error){
		testcontainersHostFromProperties,
		dockerHostFromEnv,
		dockerHostFromContext,
		dockerSocketPath,
		dockerHostFromProperties,
		rootlessDockerSocketPath,
	}

	var errs []error
	for _, dockerHostFn := range dockerHostFns {
		dockerHost, err := dockerHostFn(ctx)
		if err != nil {
			if !isHostNotFound(err) {
				errs = append(errs, err)
			}
			continue
		}

		if err = defaultCallbackCheckFn(ctx, dockerHost); err != nil {
			errs = append(errs, fmt.Errorf("check host %q: %w", dockerHost, err))
			continue
		}

		return dockerHost, nil
	}

	if len(errs) > 0 {
		return "", errors.Join(errs...)
	}

	return "", ErrSocketNotFound
}

Reply:
I like it, thanks for this suggestion. I'm adding it asap